base/src/main/java/uk/ac/ebi/atlas/resource/DataFileHub.java

```diff
@@ -91,6 +93,10 @@ public MicroarrayExperimentFiles getMicroarrayExperimentFiles(String experimentA
         return new MicroarrayExperimentFiles(experimentAccession, arrayDesign);
     }
 
+    public DifferentialExperimentFiles getReactomePathwaysCFiles(String experimentAccession, String comparison) {
```

Comment:
This isn't needed because you can use get DifferentialExperimentFiles()

Reply:
Actually not because I have a second constructor for DifferentialExperimentFiles with an extra parameter and that is the one I need, not the getDifferentialExperimentFiles

Reply:
Ah right I didn't notice the new constructor. Instead of that constructor how about adding a method to DifferentialExperimentFiles?
```
public AtlasResource<TsvReader> getReactomePathways(String comparison){
   return new TsvFile.ReadOnly(
                             dataFilesLocation, REACTOME_PATHWAYS_FILE_PATH_TEMPLATE,
                             experimentAccession, comparison);
}
```

Adding a constructor that changes what the class is for - it used to represent one experiment, now it represents experiment and a comparison? - isn't good OO (it's not just a matter of programming style), and we can do better by using the above. You can see there was a similar need in e.g. BaselineExperimentFiles that needed a different data file depending on the unit, and that was a solution.

Reply:
Good, I am making that change, it didn't come into my mind at that time. but just to clarify "overloaded constructor" is part of the OO programming, so is not a bad practice as you have referred to.

Reply:
(Y)! Yes sometimes multiple constructors are a good idea. Don't use them to change what the class represents when you only need to add one use case, that's the bad practice part.